Solving issue #23: randomising order of participants

falafel72 opened the pull request:
Added client-side javascript to allow for randomising order of leaderboard.

blazeu commented on static/js/scripts.js:
```diff
@@ -0,0 +1,11 @@
+window.onload = function() {
```
No need to put on `onload`.
The script is already on the bottom of the page.

blazeu commented on static/js/scripts.js:
```diff
@@ -0,0 +1,11 @@
+    // select all leaderboards
+    var allLeaders = document.querySelectorAll("div.orgLists ul");
+    allLeaders.forEach(function(orgList) {
+        //shuffle each leaderboard
```
space after //

blazeu reviewed (changes requested):
After you're done, run `coala` to format the code https://coala.io.

---

```
$(npm bin)/prettier --single-quote --no-semi --trailing-comma es5 --write static/js/scripts.js
```
^ if you can't run coala.

blazeu commented:
Also, keep in mind to not push changes from your master branch - create a new branch instead. Might lead to some conflicts. Refer http://api.coala.io/en/latest/Developers/Newcomers_Guide.html#step-3-creating-a-fork-and-testing-your-changes. :smile:

falafel72 commented:
I'll be sure to keep that in mind 👍 
Everything should be fixed now.

blazeu commented:
Nice. Please squash into a single commit 
https://api.coala.io/en/latest/Developers/Git_Basics.html#squashing-your-commits

blazeu reviewed (changes requested):
Format your commit message according to https://coala.io/commit

falafel72 commented:
Things got a little awkward cause I originally cloned it and forgot to fork...should be fine now though.

blazeu commented:
No, still 5 commit :/
You can ask in our chatroom @ https://coala.zulipchat.com/ if you're confused :)

blazeu commented:
Added -> Add

falafel72 commented:
Just tried using `git reset --soft`, did that work?

blazeu commented:
Run `git rebase -i HEAD~7`

7 is your commit count in this case.

Then, you want to replace `pick` to `squash`, except for the first one.

Hit save and tidy up your commit message so that it only contains your main commit.

jayvdb commented:
Comment on 9a4648c91e8742e9e93ac348c758efa8f4ad4eea, file scrap.js, line 34.

Line contains following spacing inconsistencies:
- Trailing whitespaces.

*Origin: SpaceConsistencyBear, Section: `all.whitespace`.*

The issue can be fixed by applying the following patch:

```diff
--- a/tmp/tmpo1u7h43c/scrap.js
+++ b/tmp/tmpo1u7h43c/scrap.js
@@ -31,7 +31,7 @@
 
 ;(async () => {
   const data = await fetchOrgsWithLeaders()
-  
+
   // sort data by completed_task_instance_count
   data.sort((a, b) => 
     b.completed_task_instance_count - a.completed_task_instance_count
```

jayvdb commented:
Comment on 9a4648c91e8742e9e93ac348c758efa8f4ad4eea, file scrap.js, line 36.

Line contains following spacing inconsistencies:
- Trailing whitespaces.

*Origin: SpaceConsistencyBear, Section: `all.whitespace`.*

The issue can be fixed by applying the following patch:

```diff
--- a/tmp/tmpo1u7h43c/scrap.js
+++ b/tmp/tmpo1u7h43c/scrap.js
@@ -33,7 +33,7 @@
   const data = await fetchOrgsWithLeaders()
   
   // sort data by completed_task_instance_count
-  data.sort((a, b) => 
+  data.sort((a, b) =>
     b.completed_task_instance_count - a.completed_task_instance_count
   )
 
```

jayvdb commented on package.json:
```diff
@@ -11,14 +11,16 @@
   "license": "MIT",
   "dependencies": {
     "mustache": "^2.3.0",
-    "ncp": "^2.0.0",
```
This file does not contain parsable JSON. Expecting ',' delimiter.

*Origin: JSONFormatBear, Section: `json`.*

jayvdb commented:
Comment on 9a4648c91e8742e9e93ac348c758efa8f4ad4eea, file .travis.yml, line 10.

too few spaces before comment (comments)

*Origin: YAMLLintBear, Section: `yaml`.*

jayvdb commented:
Comment on 9a4648c91e8742e9e93ac348c758efa8f4ad4eea, file .travis.yml, line 12.

truthy value is not quoted (truthy)

*Origin: YAMLLintBear, Section: `yaml`.*

falafel72 commented:
Seems like the problem is that I've got merge conflicts from merging my fork and my clone and I undo the merge every time I squash all my commits. Is there a way I can integrate the changes made by the merge into my squashed commits?

blazeu commented:
Well I did tell the wrong instruction :disappointed: 

What you could do now is back up your modified files (just yours, not the others)

Add this repo as `upstream` remote and then `git fetch upstream`

Do `git reset --hard upstream/master` (this will remove your change, so make sure you have a backup).

Then apply your changes manually and commit.

Do not push until your new commit count is exactly 1 (check `git log`).

You should push with `-f` after this.

You can ask more here https://coala.zulipchat.com

Commit:
static/js/app.js: Add randomisation of leaderboard order.

Add code which allows for randomised order of participants on
leaderboard. Lists should be shuffled on page load.

Closes https://github.com/coala/gci-leaders/issues/23

jayvdb commented on static/js/app.js:
```diff
@@ -26,3 +26,12 @@ function timeDifference(current, previous) {
+  for(var i = orgList.children.length; i >= 0; i--) {
+    orgList.appendChild(orgList.children[Math.random()*i | 0]);
+  }
+});
```
Line contains following spacing inconsistencies:
- No newline at EOF.

*Origin: SpaceConsistencyBear, Section: `all.whitespace`.*

The issue can be fixed by applying the following patch:

```diff
--- a/tmp/tmpd5h9drm4/static/js/app.js
+++ b/tmp/tmpd5h9drm4/static/js/app.js
@@ -34,4 +34,4 @@
   for(var i = orgList.children.length; i >= 0; i--) {
     orgList.appendChild(orgList.children[Math.random()*i | 0]);
   }
-});+});
```

falafel72 commented:
Thanks, looks like it's finally working.

andrewda commented:
ack 3aafa26998d5505f12385ee8728b6eb913755ceb
